crates/ruff/src/rules/pylint/rules/bad_super_call.rs

```diff
@@ -0,0 +1,103 @@
+use rustpython_parser::ast::{self, Arg, Expr, Ranged, Stmt};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_semantic::scope::ScopeKind;
+
+use crate::checkers::ast::Checker;
+
+#[violation]
+pub struct BadSuperCall {
+    base: String,
+    class: String,
+}
+
+impl Violation for BadSuperCall {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let BadSuperCall { base, class } = self;
+        format!("Bad Super Call `{base}` is not super for class `{class}`")
+    }
+}
+/// Returns `true` if a call is an argumented `super` invocation.
+fn is_super_call_with_arguments(func: &Expr, args: &[Expr]) -> bool {
+    if let Expr::Name(ast::ExprName { id, .. }) = func {
+        id == "super" && !args.is_empty()
+    } else {
+        false
+    }
+}
+
+/// PLE0241
+/// pub(crate) fn bad_super_call(checker: &mut Checker, name: &str, bases: &[Expr]) {
+pub(crate) fn bad_super_call(
+    checker: &mut Checker,
+    expr: &Expr,
+    func: &Expr,
+    args: &[Expr],
+) {
+    // Only bother going through the super check at all if we're in a `super` call.
+    // (We check this in `super_args` too, so this is just an optimization.)
+    if !is_super_call_with_arguments(func, args) {
+        return;
+    }
+    let scope = checker.ctx.scope();
+
+    // Check: are we in a Function scope?
+    if !matches!(scope.kind, ScopeKind::Function(_)) {
+        return;
+    }
+
+    let mut parents = checker.ctx.parents();
+
+    // For a `super` invocation to be unnecessary, the first argument needs to match
+    // the enclosing class, and the second argument needs to match the first
+    // argument to the enclosing function.
+    let [first_arg, second_arg] = args else {
+        return;
+    };
+    // Find the enclosing function definition (if any).
+    let Some(Stmt::FunctionDef(ast::StmtFunctionDef {
+        args: parent_args, ..
+    })) = parents
+        .find(|stmt| stmt.is_function_def_stmt()) else {
+        return;
+    };
+
+    // Extract the name of the first argument to the enclosing function.
+    let Some(Arg {
+        arg: parent_arg, ..
+    }) = parent_args.args.first() else {
+        return;
+    };
+    let Expr::Name(ast::ExprName {id: first_arg_id, ..}) = first_arg else {
+        return;
+    };
+
+    // Find the enclosing class definition (if any).
+    let Some(Stmt::ClassDef(ast::StmtClassDef {
+        name: parent_name, ..
+    })) = parents
+        .find(|stmt| matches!(stmt, Stmt::ClassDef (_))) else {
+        return;
+    };
+
+    // Need help here, do not know how to get the base_class name
+    let base_class_stmt = checker.ctx.stmt(); // ??? wrong 
+    // How do I get  the Identifier of the base class?
+    let Some(Stmt::ClassDef(ast::StmtClassDef {
+        name: base_class_id, ..
+    })) = base_class_stmt else {
+        return;
+    };
```

Comment:
This is where I have found the difficulty. I do not know how to get the `base_class` name of the current class. Otherwise it should be quite simple

Reply:
Where you have this:

```rust
// Find the enclosing class definition (if any).
let Some(Stmt::ClassDef(ast::StmtClassDef {
    name: parent_name, ..
})) = parents
    .find(|stmt| matches!(stmt, Stmt::ClassDef (_))) else {
    return;
};
```

You should be able to add the `bases`:

```rust
// Find the enclosing class definition (if any).
let Some(Stmt::ClassDef(ast::StmtClassDef {
    name: parent_name, ..
    bases: bases,
})) = parents
    .find(|stmt| matches!(stmt, Stmt::ClassDef (_))) else {
    return;
};
```